### PR TITLE
Markdown Serialization supports ordered list

### LINKED ISSUE #108: Markdown Serialization can't handle ordered lists
## Steps to Reproduce

1. Create a ParchmentDocument containing an ordered list like this:
```
1. Hello
2. This is a
3. List
```
4. Serialize it as markdown
5. see that it looks like this now: `1. Hello1. This is a1. List`

## Logs
The list also can't be deserialized anymore:
```
flutter: 'package:parchment/src/document/leaf.dart': Failed assertion: line 115 pos 12: 'value.isInline || value.isEmpty': Style cannot be applied to this leaf node: {block: ol}
leaf.dart:115
flutter: #2      LeafNode.applyStyle
leaf.dart:115
flutter: #3      LeafNode.formatAndOptimize
leaf.dart:108
flutter: #4      LineNode._insertSafe
line.dart:351
flutter: #5      LineNode.insert
line.dart:219
flutter: #6      ContainerNode.insert
node.dart:241
flutter: #7      ParchmentDocument._loadDocument
document.dart:322
flutter: #8      new ParchmentDocument.fromDelta
document.dart:64

```

### PR DESCRIPTION
**Pull request by amantoux**
Fixes #108

### CODE REVIEW

**Review by Amir-P — COMMENTED** [review]
LGTM. Tho it partially solves the #108. I have some local changes to fix that. It also changes codecs to use `ParchmentDocument` instead of `Delta` to ensure valid and working document. But it's a breaking change. WDYT?

**Comment by amantoux** [comment]
In what way is to partially fixing #108 ? Do you mind elaborating?

Re the conversion to `ParchmentDocument`, why not? we can treat that in a separate PR right?

**Comment by amantoux** [comment]
Ok tests are weak, they passed for the deserialization although they were incorrect
I'm fixing it

Re the conversion to `ParchmentDocument` I think it makes more sense indeed

**Comment by amantoux** [comment]
@Amir-P PTAL?

File: packages/parchment/lib/src/convert/markdown.dart [hunk 1]
```diff
@@ -328,6 +333,11 @@ class _ParchmentMarkdownEncoder extends Converter<Delta, String> {
         buffer.write(currentBlockLines.join('\n'));
         _writeAttribute(buffer, blockStyle, close: true);
         buffer.writeln();
+      } else if (blockStyle == ParchmentAttribute.ol) {
```
[the remark this answers is not in the document]
Comment (amantoux) [reply packages/parchment/lib/src/convert/markdown.dart:336]: Yes

File: packages/parchment/lib/src/convert/markdown.dart [hunk 2]
```diff
@@ -101,6 +101,10 @@ class _ParchmentMarkdownDecoder extends Converter<String, Delta> {
   // all blocks are supported within bq
   bool _handleBlockQuote(String line, Delta delta,
       [Map<String, dynamic>? attributes]) {
+    // we do not support nested blocks
```
[the remark this answers is not in the document]
Comment (amantoux) [reply packages/parchment/lib/src/convert/markdown.dart:104]: I'm not sure exactly why, but in any case I agree it makes more sense to use `ParchmentDocument`

**Comment by amantoux** [comment]
@Amir-P I'm fine with you comments

Can we merge this (lots of specific that needs refactoring)? It will unblock users in the short term

We can work on your proposal in another PR

**Pull request merged**